src/rsqsim_api/rsqsim_api/catalogue/catalogue.py

```diff
@@ -463,12 +489,244 @@ def events_by_number(self, event_number: Union[int, np.int, Iterable[np.int]], f
+                else:
+                    colours = np.nan * np.ones(fault.patch_numbers.shape)
+                for local_id, patch_id in enumerate(fault.patch_numbers):
+                    if patch_id in self.accumulated_slip.keys():
+                        if min_slip is not None:
+                            if self.accumulated_slip[patch_id] >= min_slip:
+                                colours[local_id] = self.accumulated_slip[patch_id]
+                        else:
+                            if self.accumulated_slip[patch_id] > 0.:
+                                colours[local_id] = self.accumulated_slip[patch_id]
+                colour_dic[f_i] = colours
+                if np.nanmax(colours) > max_slip:
+                    max_slip = np.nanmax(colours)
+        max_slip = global_max_slip if global_max_slip > 0 else max_slip
+
+        crustal_plot = None
+        for f_i, fault in enumerate(fault_model.faults):
+            if fault.name not in bruce_subduction:
+                if plot_log_scale:
+                    crustal_plot = ax.tripcolor(fault.vertices[:, 0], fault.vertices[:, 1], fault.triangles,
+                                                facecolors=colour_dic[f_i],
+                                                cmap=log_cmap, norm=colors.LogNorm(vmin=log_min, vmax=log_max))
+                else:
+                    crustal_plot = ax.tripcolor(fault.vertices[:, 0], fault.vertices[:, 1], fault.triangles,
+                                                facecolors=colour_dic[f_i],
+                                                cmap=crustal_cmap, vmin=0., vmax=max_slip)
+                plots.append(crustal_plot)
+
+        if any([subplots is None, isinstance(subplots, str)]):
+            if plot_log_scale:
+                if subduction_list:
+                    sub_cbar = fig.colorbar(subduction_plot, ax=ax)
+                    sub_cbar.set_label("Slip (m)")
+                elif crustal_plot is not None:
+                    crust_cbar = fig.colorbar(crustal_plot, ax=ax)
+                    crust_cbar.set_label("Slip (m)")
+            else:
+                if subduction_list:
+                    sub_cbar = fig.colorbar(subduction_plot, ax=ax)
+                    sub_cbar.set_label("Subduction slip (m)")
+                if crustal_plot is not None:
+                    crust_cbar = fig.colorbar(crustal_plot, ax=ax)
+                    crust_cbar.set_label("Slip (m)")
+
+        plot_coast(ax=ax)
+
+        if write is not None:
+            fig.savefig(write, dpi=300)
+            if show:
+                plt.show()
+            else:
+                plt.close(fig)
+
+        if show and subplots is None:
+            plt.show()
+
+        return plots
+
+    def plot_GR(self,fault_model: RsqSimMultiFault):
+        """
+        Plot Gutenburg-Richter distribution for a given catalogue.
+        y-axis: log(annual frequency of events with M>Mw)
```

Comment:
Change to numpy format? May require change of settings in Pycharm, I'm not sure.

Reply:
I've changed my default in Pycharm to numpy style - can regenerate them in that.